### PR TITLE
Set request parameters in Relay State for Logging and Metrics

### PR DESCRIPTION
**Pull request by saneshark**
## Description of change
Introduces:
```ruby
RelayState => {
  'signup' => 'idme', # if signing up for an account for the first time
  'verifying' => <authn_context>, # if verifying identity for the first time
  'start_time' => 1523296323, # a utc unix time to compare with saml_callback track duration
  'request_id' => '123', # the request id of the request, to track abandonment rates and track errors
}
```

Phase 1, introduce new attributes into the `RelayState`. (DONE)
Phase 2, add `sessions_controller` logging and metrics based on these attributes.
Phase 2, add Rails logging for new user signup and first time verifications.
Phase 3, add additional logging in postgresql possibly (optional)

## Testing done
local testing

## Testing planned
Grafana dashboards should tell the full picture once Phase 2 is done.

#### Unique to this PR

- [ ] _Replace this line with individual tasks unique to your PR_

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

**Comment by saneshark** [comment]
Fixes: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16842

File: app/controllers/v0/sessions_controller.rb [hunk 1]
```diff
@@ -16,57 +14,73 @@ class SessionsController < ApplicationController
     # @type is set automatically by the routes in config/routes.rb
     # For more details see SAML::SettingsService and SAML::URLService
     def new
-      type = params[:signup] ? 'signup' : params[:type]
-      if REDIRECT_URLS.include?(type)
-        url = url_service.send("#{type}_url")
-
-        # If a clientId param exists, include GA clientId for cross-domain analytics
-        client_id = params[:clientId]
```
Comment (annaswims) [inline app/controllers/v0/sessions_controller.rb]: I think this is a bad merge. it undoes the work from #2892

**Comment by annaswims** [comment]
Are each of your phases separate PR's with separate issues?

File: app/controllers/v0/sessions_controller.rb (hunk 1 above)
Comment (saneshark) [reply app/controllers/v0/sessions_controller.rb]: yea, i definitely screwed something up... i'll have to close this and cherry pick.

**Comment by saneshark** [comment]
No, i think it's going to be an epic, with multiple linked issues.

**Comment by saneshark** [comment]
closing this for now

**Pull request closed without merging**